app/src/main/java/com/imdb_compose/ui/PersonDetailsPage.kt

```diff
@@ -0,0 +1,176 @@
+package com.imdb_compose.ui
+
+import android.annotation.SuppressLint
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Add
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonColors
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewModelScope
+import androidx.navigation.NavController
+import coil.compose.AsyncImage
+import com.imdb_compose.BottomBar
+import com.imdb_compose.domain.Retrofit
+import com.imdb_compose.TopBarWithBackBtn
+import com.imdb_compose.gray500
+import com.imdb_compose.isLoading
+import com.imdb_compose.ripeMango
+import com.imdb_compose.shadow
+import kotlinx.coroutines.launch
+
+@SuppressLint("CoroutineCreationDuringComposition", "StateFlowValueCalledInComposition")
+@Composable
+fun PersonDetailsPage(
+    person: String,
+    id: Int,
+    viewModel: HomeScreenViewModel,
+    navController: NavController,
+    clickHandlerBackBtn: () -> Unit
+) {
+    Scaffold(
+        modifier = Modifier.fillMaxSize(),
+        containerColor = Color.Transparent,
+        topBar = {
+            TopBarWithBackBtn(person, clickHandlerBackBtn)
+        },
+        bottomBar = {
+            BottomBar(navController)
+        }
+    ) { paddingValues ->
+        viewModel.viewModelScope.launch {
+            viewModel.getPersonDetails(id)
+        }
+
+        if (viewModel.personDetails.value?.id == id) {
```

Comment:
I see this is your test if the request is complete, but it's kind of clunky. For one, what if you get an error? How do you differentiate between not having the object because it's still loading or if it errored out. This is a good sign you need something like `Async` to keep track of the state of your requests